### PR TITLE
addition(STONEINTG-501): explain image extraction details

### PR DESCRIPTION
**Pull request by jsztuka**
Goal of this addition is to explain how IS(Integration Service) is extracting image information from build pipeline run.

### CODE REVIEW

**Comment by jsztuka** [comment]
Asking @ralphbean and @arewm for a review on this one. Thanks!

File: book/integration-service.md [hunk 1]
```diff
@@ -205,6 +205,10 @@ The Integration service needs secrets mounted so that the `Environment Provision
 16. Repeat step 10
 17. Repeat step 11-13
 
+### Image extraction details
+
+Integration service is getting specific information about image being build by build-definition using [those functions](https://github.com/redhat-appstudio/integration-service/blob/06a2947061c943ea9a28a1a39957737fcf8bdc92/tekton/utils.go#L100-L146). Each one of them corresponds with the [results](https://github.com/redhat-appstudio/build-definitions/blob/5881b7381ec2efe226ddcfb6c19a8a9e71e68166/pipelines/template-build/template-build.yaml#L274-L281) that are being reported by build-definitions.
```
Comment (dirgim) [inline book/integration-service.md:210]: Maybe we can list the result names and what they represent? 
We can also have a blurb on how the Integration service uses those results - e.g. the `IMAGE_URL` and `IMAGE_DIGEST` are used to construct the component image reference, and `CHAINS-GIT_URL` and `CHAINS-GIT_COMMIT` are added to the source section of the component within the Snapshot.

Comment (jsztuka) [reply book/integration-service.md:210]: Agreed, will include results with their usage in IS.

File: book/integration-service.md [hunk 2]
```diff
@@ -205,6 +205,20 @@ The Integration service needs secrets mounted so that the `Environment Provision
 16. Repeat step 10
 17. Repeat step 11-13
 
+### Image extraction details
+
+Integration service is getting specific information about image being build by build-definition using individual functions. Each one of them corresponds with the results that are being reported by build-definitions.
```
Comment (dirgim) [inline book/integration-service.md:210]: Maybe something along these lines would make it more clear:
```suggestion
Integration service is getting specific information about the image that's being built by the RHTAP build pipeline by parsing the expected Tekton results for the pipeline. All of them are required to be present in order to correctly construct a Snapshot.
```

File: book/integration-service.md [hunk 3]
```diff
@@ -205,6 +205,20 @@ The Integration service needs secrets mounted so that the `Environment Provision
...
+
+<ins>Results are following:</ins>
+ - IMAGE_URL
```
Comment (dirgim) [inline book/integration-service.md:213]: Maybe we can have a short description for each result and then mention what each of them is used for - e.g. for this one we could say something like this:
```
Image repository where the built image was pushed. Used to construct the component image reference.
```

**Review by ralphbean — APPROVED** [review]

**Review by dirgim — APPROVED** [review]
LGTM

**Pull request merged**